src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -213,22 +220,20 @@ protected override string BuildChallengeUrl([NotNull] AuthenticationProperties p
 
         if (Options.UsePkce)
         {
-            var bytes = RandomNumberGenerator.GetBytes(32);
-            var codeVerifier = Microsoft.AspNetCore.Authentication.Base64UrlTextEncoder.Encode(bytes);
+            byte[] bytes = RandomNumberGenerator.GetBytes(32);
```

Comment:
I can change it, but the 32 was lifted from [`OAuthHandler<T>`](https://github.com/dotnet/aspnetcore/blob/de8ee82692989d46d54d1e1be2d3254e4436d720/src/Security/Authentication/OAuth/src/OAuthHandler.cs#L306).

Reply:
We could use `HMACSHA256.HashSizeInBits` instead when 7 lands: https://github.com/dotnet/runtime/pull/63685

Reply:
We could: technically, it would do exactly same thing. But semantically, it would be super weird: `HashSizeInBits` represents the size of the message digest (i.e the output of the hash method) while the random array we generate here is used as the message (i.e the input given to the hash method). SHA256 can accept inputs of any length up to 2^64-1 and there's no relationship between the size of the input and the output so it would be weird to mix the two things.

FWIW, 256 comes from the PKCE spec itself, that recommends generating code verifiers with at least 256 bits of entropy: https://datatracker.ietf.org/doc/html/rfc7636#section-7.1 (256 bits of entropy already give a very high level of security so not sure why anyone would want to opt for a higher value 😃)